CONTRIBUTING.md

```diff
@@ -1,35 +1,118 @@
+# Contributing
+
+Thanks for considering to contribute to this project. Please follow the below guidelines before you
+open up a pull request.
+
+## Nomenclature
+
+First a brief intro on some of the terminoligy and concepts we use to make sure the steps below make sense.
```

Comment:
Can you add a `,` to make it more readable :
First a brief intro on some of the terminoligy and concepts we use, to make sure the ....

btw at `terminoligy` my auto correct says it terminology